packages/definitions-parser/test/git.test.ts

```diff
@@ -21,43 +21,104 @@ const deleteJestDiffs: GitDiff[] = [
+  { status: "D", file: "types/rdfjs__environment/NamespaceFactory.d.ts" },
+  { status: "D", file: "types/rdfjs__environment/TermMapSetFactory.d.ts" },
+  { status: "M", file: "types/rdfjs__environment/package.json" },
+  { status: "M", file: "types/rdfjs__environment/rdfjs__environment-tests.ts" },
+  { status: "R", source: "types/rdfjs__formats-common/.npmignore", file: "types/rdfjs__formats/.npmignore" },
+  { status: "R", source: "types/rdfjs__environment/FormatsFactory.d.ts", file: "types/rdfjs__formats/Factory.d.ts" },
+  { status: "R", source: "types/rdfjs__formats-common/index.d.ts", file: "types/rdfjs__formats/index.d.ts" },
+  { status: "R", source: "types/rdfjs__environment/lib/Formats.d.ts", file: "types/rdfjs__formats/lib/Formats.d.ts" },
+  { status: "R", source: "types/rdfjs__formats-common/package.json", file: "types/rdfjs__formats/package.json" },
+  { status: "A", file: "types/rdfjs__formats/pretty.d.ts" },
+  {
+    status: "R",
+    source: "types/rdfjs__formats-common/rdfjs__formats-common-tests.ts",
+    file: "types/rdfjs__formats/rdfjs__formats-tests.ts",
+  },
+  { status: "A", file: "types/rdfjs__formats/tsconfig.json" },
+  { status: "M", file: "types/rdfjs__serializer-jsonld-ext/index.d.ts" },
+  { status: "M", file: "types/rdfjs__serializer-jsonld-ext/package.json" },
+  { status: "M", file: "types/rdfjs__serializer-jsonld-ext/rdfjs__serializer-jsonld-ext-tests.ts" },
+  { status: "A", file: "types/rdfjs__serializer-turtle/.npmignore" },
+  { status: "A", file: "types/rdfjs__serializer-turtle/index.d.ts" },
+  { status: "A", file: "types/rdfjs__serializer-turtle/package.json" },
+  { status: "A", file: "types/rdfjs__serializer-turtle/rdfjs__serializer-turtle-tests.ts" },
+  {
+    status: "R",
+    source: "types/rdfjs__formats-common/tsconfig.json",
+    file: "types/rdfjs__serializer-turtle/tsconfig.json",
+  },
+];
+function getDeletions(diffs: GitDiff[]): PackageId[] {
+  const changes = gitChanges(diffs);
+  expect(changes).not.toHaveProperty("error");
+  const { deletions } = changes as { deletions: PackageId[]; additions: PackageId[] };
+  return deletions;
+}
 
 testo({
   async ok() {
-    expect(await getNotNeededPackages(allPackages, deleteJestDiffs)).toEqual(jestNotNeeded);
+    expect(await getNotNeededPackages(allPackages, getDeletions(deleteJestDiffs))).toEqual(jestNotNeeded);
   },
+  async gitMovesConvertedToAddsAndDeletes() {
+    expect(gitChanges(moveRdfJSDiffs)).toEqual({
+      // serializer-jsonld-ext has changes, why isn't in the list? I guess we only track dirs with additions and deletions?
+      additions: [
+        { typesDirectoryName: "rdf-ext", version: "*" },
+        { typesDirectoryName: "rdfjs__formats", version: "*" },
+        { typesDirectoryName: "rdfjs__serializer-turtle", version: "*" },
+      ],
+      deletions: [
+        { typesDirectoryName: "rdfjs__environment", version: "*" }, // this still has stuff in it, why is it put in deletions?
```

Comment:
RE: this comment and the other comment above, the diffs only track what changed, yeah. I think that if anything, these could be named "potential additions" and "potential deletions" and we already run tests assuming those are the case?

Maybe this means that we are missing some check for "deleted some of the files but not all" where it needs to at least see if the package seems like it exists... but I think that's something you already fixed?

Reply:
Oh, I need to remove this question; it is indeed "packages *with* deleted *files*", not "deleted *packages*".